### ISSUE TITLE
Client does not check result of blade operations for errors

### ISSUE THREAD

**Issue by crienzo** [signalwire/signalwire-ruby-old#41]
Running ruby client and got errors when attempting blade.execute on signalwire.setup and call.receive and blade.subscription on notification channel.  These errors should be checked.  They are not so the client believes it has successfully connected to SignalWire.

```
<13>:ruby tts.rb
I, [2019-08-11T09:02:14.547971 #19562]  INFO -- SignalWire: Starting up TTSConsumer
D, [2019-08-11T09:02:14.548228 #19562] DEBUG -- SignalWire: Connecting to wss://relay.signalwire.com
D, [2019-08-11T09:02:14.548363 #19562] DEBUG -- SignalWire: Running with epoll true
D, [2019-08-11T09:02:14.869401 #19562] DEBUG -- SignalWire: SEND: {
  "method": "blade.connect",
  "params": {
    "version": {
      "major": 2,
      "minor": 1,
      "revision": 0
    },
    "authentication": {
      "project": "xxxxxxxxxx",
      "token": "xxxxxxxxxxx"
    }
  },
  "jsonrpc": "2.0",
  "id": "e674cd4f-bed9-434f-895a-9e6002ae4761"
}
D, [2019-08-11T09:02:14.923471 #19562] DEBUG -- SignalWire: RECV: {
  "jsonrpc": "2.0",
  "id": "e674cd4f-bed9-434f-895a-9e6002ae4761",
  "result": {
    "session_restored": false,
    "sessionid": "c92d57b6-d2fc-4d4f-b216-2a1f797a69dc",
    "nodeid": "82f8d536-9558-4bba-bc1f-d76d760ba16e",
    "master_nodeid": "00000000-0000-0000-0000-000000000000",
    "authorization": {
      "project": "xxxxxxxxxxxx",
      "expires_at": null,
      "scopes": [
        "calling",
        "messaging"
      ],
      "signature": "xxxxxxxxxx"
    },
    "routes": [

    ],
    "protocols": [

    ],
    "subscriptions": [

    ],
    "authorities": [

    ],
    "authorizations": [

    ],
    "accesses": [

    ],
    "protocols_uncertified": [
      "signalwire"
    ]
  }
}
I, [2019-08-11T09:02:14.943740 #19562]  INFO -- SignalWire: Blade Session connected with id: c92d57b6-d2fc-4d4f-b216-2a1f797a69dc
D, [2019-08-11T09:02:14.943881 #19562] DEBUG -- SignalWire: Relay client connected
D, [2019-08-11T09:02:14.949873 #19562] DEBUG -- SignalWire: SEND: {
  "method": "blade.execute",
  "params": {
    "protocol": "signalwire",
    "method": "setup",
    "params": {
    }
  },
  "jsonrpc": "2.0",
  "id": "fbf5def2-134a-4097-83cb-7fb723639831"
}
D, [2019-08-11T09:02:15.008500 #19562] DEBUG -- SignalWire: RECV: {
  "jsonrpc": "2.0",
  "id": "fbf5def2-134a-4097-83cb-7fb723639831",
  "error": {
    "requester_nodeid": "82f8d536-9558-4bba-bc1f-d76d760ba16e",
    "responder_nodeid": "ae6970c1-b3c6-437a-8d44-fad9e65424d3",
    "code": -32003,
    "message": "Certified clients are not permitted to setup relay protocols"
  }
}
D, [2019-08-11T09:02:15.008917 #19562] DEBUG -- SignalWire: Protocol set up as 
D, [2019-08-11T09:02:15.009358 #19562] DEBUG -- SignalWire: SEND: {
  "method": "blade.subscription",
  "params": {
    "protocol": null,
    "command": "add",
    "channels": [
      "notifications"
    ]
  },
  "jsonrpc": "2.0",
  "id": "892cc01d-6ca0-4e96-aaec-4ec261d67ab0"
}
D, [2019-08-11T09:02:15.054812 #19562] DEBUG -- SignalWire: RECV: {
  "jsonrpc": "2.0",
  "id": "892cc01d-6ca0-4e96-aaec-4ec261d67ab0",
  "error": {
    "code": -32602,
    "message": "Invalid SubscriptionParams"
  }
}
D, [2019-08-11T09:02:15.055347 #19562] DEBUG -- SignalWire: Subscribed to notifications for 
D, [2019-08-11T09:02:15.056154 #19562] DEBUG -- SignalWire: SEND: {
  "method": "blade.execute",
  "params": {
    "protocol": null,
    "method": "call.receive",
    "params": {
      "context": "tts"
    }
  },
  "jsonrpc": "2.0",
  "id": "6ad7d15f-2745-43ad-8831-c01f0ecead43"
}
D, [2019-08-11T09:02:15.103992 #19562] DEBUG -- SignalWire: RECV: {
  "jsonrpc": "2.0",
  "id": "6ad7d15f-2745-43ad-8831-c01f0ecead43",
  "error": {
    "code": -32602,
    "message": "Invalid ExecuteParams"
  }
}
E, [2019-08-11T09:02:15.104731 #19562] ERROR -- SignalWire: Unknown Relay command failure, result code not found
I, [2019-08-11T09:02:15.105067 #19562]  INFO -- SignalWire: Connected to SignalWire!
 ^CI, [2019-08-11T09:04:23.572414 #19562]  INFO -- SignalWire: TTSConsumer Consumer shutting down.
```

**Comment by bryanrite** [comment]
@lpradovera this is a priority. I'd like to get this in with all the other fixes and release a new patch-level ruby SDK tomorrow.

**Comment by lpradovera** [comment]
Fixed in #42
